bin/check-aggregate-output.rb

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env ruby
+#
+# Check Aggregate Output
+# ===
+# This check is very similar to check-aggregate.rb, it checks the Sensu aggregate
+# api and returns Critical / Warning / OK based on how many aggregate results
+# match the options you configure.
+#
+# Authors
+# ===
+# AJ Bourg, @ajbourg
+# Sean Porter, @portertech
+#
+# Copyright 2017
+#
+# Released under the same terms as Sensu (the MIT license); see
+# LICENSE for details.
+
+require 'sensu-plugin/check/cli'
+require 'rest-client'
+require 'json'
+
+class CheckAggregateOutput < Sensu::Plugin::Check::CLI
+  option :api,
+         short: '-a URL',
+         long: '--api URL',
+         description: 'Sensu API URL',
+         default: if ENV['SENSU_API']
+                    ENV['SENSU_API'] + ':4567'
+                  elsif ENV['SENSU_API_URL']
+                    ENV['SENSU_API_URL']
+                  else
+                    'http://localhost:4567'
+                  end
+
+  option :insecure,
+         short: '-k',
+         boolean: true,
+         description: 'Enabling insecure connections',
+         default: false
+
+  option :user,
+         short: '-u USER',
+         long: '--user USER',
+         description: 'Sensu API USER'
+
+  option :password,
+         short: '-p PASSWORD',
+         long: '--password PASSWORD',
+         description: 'Sensu API PASSWORD'
+
+  option :timeout,
+         short: '-t SECONDS',
+         long: '--timeout SECONDS',
+         description: 'Sensu API connection timeout in SECONDS',
+         proc: proc(&:to_i),
+         default: 30
+
+  option :check,
+         short: '-c CHECK',
+         long: '--check CHECK',
+         description: 'Aggregate CHECK name',
+         required: true
+
+  option :age,
+         short: '-A SECONDS',
+         long: '--age SECONDS',
+         description: 'Minimum aggregate age in SECONDS, time since check request issued',
+         default: 30,
+         proc: proc(&:to_i)
+
+  option :message,
+         short: '-M MESSAGE',
+         long: '--message MESSAGE',
+         description: 'A custom error MESSAGE'
+
+  option :collect_output,
+         short: '-o',
+         long: '--output',
+         boolean: true,
+         description: 'Collects all non-ok outputs',
+         default: true
+
+  option :warning,
+         short: '-W PERCENT',
+         long: '--warning PERCENT',
+         description: 'PERCENT warning before warning (can be change with --ignore-severity)',
+         proc: proc(&:to_i)
+
+  option :warning_count,
```

Comment:
While I realize this is following the existing pattern it might be a good time to make some would be breaking changes and make this a bit better. Rather than having four options I believe we can reduce this by having `--warning`, `--critical`, and something like `--alert-on` to specify count or percentages. I would suggest default of percentages as we can make better sane defaults for small and larger setups. This reduces the required knobs to get the check working.